tests/test_rich_import.py

```diff
@@ -0,0 +1,21 @@
+import subprocess
+import sys
+from pathlib import Path
+
+ACCEPTED_MODULES = {"rich._extension", "rich"}
+
+
+def test_rich_not_imported_unnecessary():
+    file_path = Path(__file__).parent / "assets/print_modules.py"
+    result = subprocess.run(
+        [sys.executable, "-m", "coverage", "run", str(file_path)],
+        capture_output=True,
+        encoding="utf-8",
+    )
+    modules = result.stdout.splitlines()
+    modules = [
+        module
+        for module in modules
+        if module not in ACCEPTED_MODULES and module.startswith("rich")
+    ]
+    assert not modules
```

Comment:
This is a matter of personal preference, but for clarity of code I would prefer 
```suggestion
    assert len(modules) == 0
```

Reply:
This is arguably the most idiomatic style, see also [PEP8](https://peps.python.org/pep-0008/#programming-recommendations):

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false:

```python
# Correct:
if not seq:
if seq:
# Wrong:
if len(seq):
if not len(seq):
```

But I can change it if you prefer.

Reply:
I'll leave the final decision with Tiangolo 🙏

Reply:
I'm not reading this with the same interpretation. I'm not suggesting to check truthness of `len(x)`, I'm suggesting to explicitely check the length being zero, as suggested on the page you cited:
> You can either remove the call to len or **compare the length against a scalar**.